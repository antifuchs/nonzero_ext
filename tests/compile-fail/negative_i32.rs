#[macro_use]
extern crate nonzero_ext;

use std::num::NonZeroU32;

#[cfg_attr(rustfmt, rustfmt_skip)]
fn main() {
    let _a: NonZeroU32 = nonzero!(-2i32); //~ ERROR no method named
    //                   ^ should not be able to convert
}
