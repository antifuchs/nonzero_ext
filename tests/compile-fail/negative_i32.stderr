error[E0599]: no method named `as_nonzero_unchecked` found for type `i32` in the current scope
 --> $DIR/negative_i32.rs:8:26
  |
8 |     let _a: NonZeroU32 = nonzero!(-2i32);
  |                          ^^^^^^^^^^^^^^^ method not found in `i32`
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
